### PR TITLE
Allow users to provide copying of optional files without failing.

### PR DESCRIPTION
**Pull request by aerorahul**
**Description**
This PR extends the `FileHandler` `copy` method to allow users to specify if files to be copied are either `required` or `optional`.

This is done by introducing `copy_req` and `copy_opt`.  By default `copy == copy_req`.

The use is same as before; simply use `copy_opt` instead of `copy` for copying optional files.  If a file is not found, a warning will be presented.  For `copy` or `copy_req`, the error will be fatal.

This PR additionally allows users to checkout and develop `wxflow` in a vscode workspace environment.

**Type of change**

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

**How Has This Been Tested?**

Tests have been added to the extended functionality 

- [X] pynorms
- [X] pytests

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (CoryMartin-NOAA) [inline .vscode/launch.json:2]: are these vscode files supposed to be committed?

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../src')))
```
Comment (CoryMartin-NOAA) [inline tests/conftest.py:4]: is this file needed?

**Review by CoryMartin-NOAA — COMMENTED** [review]
Questions on possible stray files but the rest looks good

File: .vscode/launch.json (hunk 1 above)
Comment (aerorahul) [reply .vscode/launch.json:2]: They should be. There is nothing here that is user specific.

File: src/wxflow/file_utils.py [hunk 3]
```diff
@@ -17,8 +18,10 @@ class FileHandler:
 
     NOTE
     ----
-    "action" can be one of mkdir", "copy", etc.
+    "action" can be one of "mkdir", "copy", "copy_req", "copy_opt", etc.
     Corresponding "act" would be ['dir1', 'dir2'], [['src1', 'dest1'], ['src2', 'dest2']]
+    "copy_req" will raise an error if the source file does not exist
+    "copy_opt" will not raise an error if the source file does not exist
```
Comment (DavidNew-NOAA) [inline src/wxflow/file_utils.py]:
```suggestion
    "copy_opt" will not raise an error if the source file does not exist but will present a warning
```

File: src/wxflow/file_utils.py [hunk 4]
```diff
@@ -35,15 +38,25 @@ def sync(self):
         Method to execute bulk actions on files described in the configuration
         """
         sync_factory = {
-            'copy': self._copy_files,
+            'copy': self.copy_req,
+            'copy_req': self.copy_req,
```
Comment (DavidNew-NOAA) [inline src/wxflow/file_utils.py:47]: ```copy``` and ```copy_req``` seems redundant since their behavior is the same. Why not just have ```copy```?

**Review by DavidNew-NOAA — COMMENTED** [review]
Looks good. See comments

File: tests/conftest.py (hunk 2 above)
Comment (aerorahul) [reply tests/conftest.py:4]: Yes.  It tells the package where the src is in relation to the tests.  Some packages do not keep code under `src/app`, but directly at `app/`

Another use for this file is to provide test fixtures that are/could be used across tests.  Eg. if `test_a.py` and `test_c.py` needs something that is common, it can be a `pytest` fixture defined in `conftest.py`.

File: src/wxflow/file_utils.py (hunk 4 above)
Comment (aerorahul) [reply src/wxflow/file_utils.py:47]: I kept the distinction for clarity.  If it were my choice, I would deprecate `copy` and make it explicit `copy_req` so it becomes clear.
Currently, we need to keep `copy` since that will break existing use.

Comment (DavidNew-NOAA) [reply src/wxflow/file_utils.py:47]: In that case, I would suggest comments to indicate that ```copy``` is moving towards deprecation and shouldn't be used in the future.

Comment (aerorahul) [reply src/wxflow/file_utils.py:47]: Sounds good.  If that is the consensus from the team, we can make that happen.
I will make a comment that `copy` will be deprecated by XYZ, and users should use `copy_req` instead.

Comment (aerorahul) [reply src/wxflow/file_utils.py:47]: done @DavidNew-NOAA

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by DavidNew-NOAA — APPROVED** [review]

**Review by DavidHuber-NOAA — APPROVED** [review]
Looks good, thanks @aerorahul!
